common/import_.py

```diff
@@ -245,93 +275,97 @@ def new_exchange(activity, new_activity, new_amount=None, activity_to_copy_from=
 
     Example : ingredient flour-organic is not in agribalyse so it is created at this step. It's a
     variant of activity flour
     """
-    activity = search(activity_data["searchIn"], activity_data["search"])
+    # Example : the flour-conventional
+    existing_activity = search(
+        activity_data["database"], activity_data["existingActivity"]
+    )
 
-    # create a new variant activity
+    # create a new  activity
     # Example: this is where we create the flour-organic activity
-    activity_variant = create_activity(
+    new_activity = create_activity(
         dbname,
-        f"{activity['name']} {activity_data['suffix']}",
-        activity,
+        f"{activity_data['newName']}",
+        existing_activity,
     )
 
-    # if the activity has no subactivities, we can directly replace the seed activity with the seed
-    #  activity variant
-    if not activity_data["subactivities"]:
-        replace_activities(activity_variant, activity_data, activity_data["searchIn"])
+    if "delete" in activity_data:
+        for activity_name in activity_data["delete"]:
+            activity_to_delete = search_activity(
+                activity_data["database"], f"{activity_name}"
+            )
+            delete_exchange(new_activity, activity_to_delete)
 
-    # else we have to iterate through subactivities and create a new variant activity for each subactivity
+    if "replace" in activity_data:
+        # if the activity has no subactivities, we can directly replace the seed activity with the seed
+        #  activity variant
+        if not activity_data["subactivities"]:
+            replace_activities(new_activity, activity_data, activity_data["database"])
 
-    # Example: for flour-organic we have to dig through the `global milling process` subactivity before
-    #  we can replace the wheat activity with the wheat-organic activity
-    else:
-        for i, act_sub_data in enumerate(activity_data["subactivities"]):
-            searchdb = None  # WARNING : the last database specified in "subactivities" is used in the "replace"
-            if isinstance(act_sub_data, list):
-                searchdb, act_sub_data = act_sub_data
-            else:
-                searchdb, act_sub_data = (
-                    searchdb or activity_data["searchIn"],
-                    act_sub_data,
+        # else we have to iterate through subactivities and create a new variant activity for each subactivity
+
+        # Example: for flour-organic we have to dig through the `global milling process` subactivity before
+        #  we can replace the wheat activity with the wheat-organic activity
+        else:
+            for i, act_sub_data in enumerate(activity_data["subactivities"]):
+                if "::" in act_sub_data:
+                    searchdb, act_sub_data = act_sub_data.split("::")
+                else:
+                    searchdb = activity_data["database"]
```

Comment:
It looks like the code to extract the db name using `::` could be extracted in a separate function as it used in multiple places.

Maybe something like:

```python
def get_db_and_activity_name(full_activity_name: str, default_db: str | None = None) -> tuple[str, str]:
```

It would do the split, check that the default_db is provided is there is no `::` and throw an exception otherwise. You could even add a test juste for this extraction function.